### ISSUE TITLE
CI Flakiness

### ISSUE THREAD

**Issue by jasnell** [nodejs/build#852]
CI has been flaky all week. Access has been timing out. I've been seeing gateway errors. Accessing the website is super slow. Is there a specific problem that needs to be looked at? The failures are interfering with being able to get work done.

**Comment by refack** [comment]
ping @nodejs/build (specifically infra team)

**Comment by refack** [comment]
@mhdawson @jbergstroem Hoping you're online

**Comment by rvagg** [comment]
Sorry, as I said on IRC, I started working on this a couple of hours ago but then went and had breakfast and forgot all about it. It is Saturday after all!

CI has been restarted and seems to be responsive. I did some poking around and: CPU was reasonable although Java was spiking way up and then way down again—although I don't believe this is abnormal for Jenkins. Disk space is great (thanks to jenkins-workspace machines fixing that!). Without anything further to go on my guess is simply that it's Jenkins fatigue for not being restarted in a while. It does so much work that the JVM memory space is _massive_ and it seems to get bogged down after a while and a restart does it wonders. It's either that or maybe an underlying problem with the hardware, maybe disk I/O issues. Will monitor further.

Will be working on ensuring all the nodes reconnect properly over the next 30 mins to an hour. Should all be good to use if you want to submit new jobs now though.

**Comment by jasnell** [comment]
Thanks @rvagg, I appreciate it. 

I think it's past due time for me to dig in to Jenkins so you don't have to spend your Saturday mornings chasing that down.

**Comment by rvagg** [comment]
@nodejs/build I've taken the Odroid C2's out of node-test-commit-arm, they are all down. The second one has known disk problems and a card needs to be replaced, the other two are unresponsive. I've reached out to David @ miniNodes to look into it, hopefully it's just a hard reboot for them.

Which actually makes me suspect that this might be part of the cause of the problems with Jenkins today—if machines are down and we get a backup of queued jobs for that type of node then Jenkins can go a little bit crazy holding on to them. So look at the queue, see if there's any class that's fully offline and work back from there. I've forced stopped the queued C2 jobs already.
